lib/trento/application/usecases/hosts/hosts.ex

```diff
@@ -36,4 +47,41 @@ defmodule Trento.Hosts do
         subscription_count
     end
   end
+
+  @spec deregister_host(Ecto.UUID.t(), DateService) :: :ok | {:error, any}
+  def deregister_host(host_id, date_service \\ DateService) do
+    case Repo.get_by(HostReadModel, id: host_id) do
+      nil ->
+        {:error, :host_not_found}
```

Comment:
Maybe better `host_not_registered` ?
This is what we use in the fallback controller as well.
This way you don't need to handle this in the controller itself